packages/hub/services/search-index.ts

```diff
@@ -259,17 +261,23 @@ class IndexerRun implements IndexerHandle {
     let componentMetaModule = definedCard.componentInfos[format].metaModule.global;
     let componentMeta = await this.fileCache.loadModule(componentMetaModule);
 
-    let cardModel = await getOwner(this).instantiate(CardModelForHub, {
-      type: 'loaded',
-      id: rawCard.id,
-      realm: rawCard.realm,
-      format,
-      rawData: rawCard.data ?? {},
-      schemaModule: definedCard.schemaModule.global,
-      componentModule: definedCard.componentInfos[format].componentModule.global,
-      componentMeta,
-      allFields: false,
-    });
+    let cardModel = await getOwner(this).instantiate(
+      CardModelForHub,
+      await this.cardService.as(INSECURE_CONTEXT),
+      {
+        type: 'loaded',
+        id: rawCard.id,
+        componentModule: definedCard.componentInfos[format].componentModule.global,
```

Comment:
I wonder if we really need to set this, does it ever change? if we don't need to set it, it would make things simpler since there would be no need to use `compiled` and the indexing could get pushed into the card service